misc/lib-creator/src/main/java/org/ballerinalang/stdlib/utils/GenerateBalo.java

```diff
@@ -196,12 +200,11 @@ public void saveCompiledPackage(CompiledPackage compiledPackage,
     }
 
     private static boolean isIsolatedWarningLog(DiagnosticCode code) {
-        switch (code) {
-            case FUNCTION_CAN_BE_MARKED_ISOLATED:
-            case WARNING_INVALID_MUTABLE_ACCESS_AS_RECORD_DEFAULT:
-            case WARNING_INVALID_NON_ISOLATED_INVOCATION_AS_RECORD_DEFAULT:
-            case WARNING_INVALID_NON_ISOLATED_INIT_EXPRESSION_AS_RECORD_DEFAULT:
-                return true;
+        if (FUNCTION_CAN_BE_MARKED_ISOLATED.equals(code) || 
+                WARNING_INVALID_MUTABLE_ACCESS_AS_RECORD_DEFAULT.equals(code) || 
+                WARNING_INVALID_NON_ISOLATED_INVOCATION_AS_RECORD_DEFAULT.equals(code) || 
+                WARNING_INVALID_NON_ISOLATED_INIT_EXPRESSION_AS_RECORD_DEFAULT.equals(code)) {
+            return true;
```

Comment:
Checking the equality of each code one-by-one can be slightly slower than the switch case.

Can we check for `instanceof DiagnosticWarningCode` and cast to it, and then use the existing switch case?